test/integration/ci-operator-config-mirror.sh

```diff
@@ -2,15 +2,23 @@
 source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
 trap os::test::junit::reconcile_output EXIT
 
-suite_dir="${OS_ROOT}/test/integration/ci-operator-config-mirror/"
+suite_dir="${OS_ROOT}/test/integration/ci-operator-config-mirror"
 cp -r "${suite_dir}/input" "${BASETMPDIR}"
+cp -r "${suite_dir}/input-to-clean" "${BASETMPDIR}"
+
 actual="${BASETMPDIR}/input"
+actual_to_clean="${BASETMPDIR}/input-to-clean"
+
 expected="${suite_dir}/output"
 
 os::test::junit::declare_suite_start "integration/ci-operator-config-mirror"
 # This test validates the ci-operator-config-mirror tool
 
-os::cmd::expect_success "ci-operator-config-mirror --to-org super-priv --config-path ${actual} --whitelist-file ${suite_dir}/whitelist.yaml"
+os::cmd::expect_success "ci-operator-config-mirror --to-org super-priv --config-path ${actual} --clean=false --whitelist-file ${suite_dir}/whitelist.yaml"
 os::integration::compare "${actual}" "${expected}"
 
+os::cmd::expect_success "ci-operator-config-mirror --to-org super-priv --config-path ${actual_to_clean} --clean=true --whitelist-file ${suite_dir}/whitelist.yaml"
+os::integration::compare "${actual_to_clean}" "${expected}"
```

Comment:
This does not seem to test the functionality though. The input and output `super-priv` are identical, so this test does not show the tool actually removing anything:

```
❯ diff -rq input-to-clean/super-priv/ output/super-priv/

test/integration/ci-operator-config-mirror on  deprecate-configs [$] 
❯ echo $status
0
```

We should have both 1) a full directory in original `super-priv` that gets removed and 2) a single file that gets removed from a directory where other files are kept. This way we actually test that the tool does what it claims.

Reply:
addressed